packages/perseus/src/widgets/radio/base-radio.jsx

```diff
@@ -17,264 +16,75 @@ import {scrollElementIntoView} from "../../util/scroll-utils.js";
 import ChoiceNoneAbove from "./choice-none-above.jsx";
 import Choice from "./choice.jsx";
 
+import type {PerseusRadioWidgetOptions} from "../../perseus-types";
+import type {APIOptions} from "../../types.js";
 import type {StyleDeclaration} from "aphrodite";
 
 const {captureScratchpadTouchStart} = Util;
 
-const ChoicesType = PropTypes.arrayOf(
-    PropTypes.shape({
-        // Indicates whether this choice is checked.
-        checked: PropTypes.bool,
-
-        // Indicates whether the user has "crossed out" this choice, meaning
-        // that they don't think it's correct. This value does not affect
-        // scoring or other behavior; it's just a note for the user's
-        // reference.
-        crossedOut: PropTypes.bool,
-
-        content: PropTypes.node,
-        rationale: PropTypes.node,
-        hasRationale: PropTypes.bool,
-        showRationale: PropTypes.bool,
-        showCorrectness: PropTypes.bool,
-        correct: PropTypes.bool,
-        originalIndex: PropTypes.number,
-        isNoneOfTheAbove: PropTypes.bool,
-    }),
-);
+type ChoiceType = {|
+    checked: boolean,
+    crossedOut: boolean,
+    content: React.Node,
+    rationale: React.Node,
+    hasRationale: boolean,
+    showRationale: boolean,
+    showCorrectness: boolean,
+    correct: boolean,
+    originalIndex: number,
+    isNoneOfTheAbove: boolean,
+    highlighted: boolean,
+    previouslyAnswered: boolean,
+    revealNoneOfTheAbove: boolean,
+    disabled: boolean,
+|};
+
+type Props = {|
+    apiOptions: APIOptions,
+    choices: ChoiceType[],
```

Comment:
**style nit (non-blocking):** Perseus uses `Array<>` and `$ReadOnlyArray<>` with a preference for `$ReadOnlyArray<>` unless a mutable array is absolutely necessary.